lib/Sema/SemaExpr.cpp

```diff
@@ -12466,6 +12466,63 @@ ExprResult Sema::CreatePositionalParameterExpr(unsigned Index, QualType QT) {
   return new (Context) PositionalParameterExpr(Index, QT);
 }
 
+bool Sema::CheckBoundsCastBaseType(Expr *E1) {
+  bool Result = false;
+  QualType SrcTy = E1->getType();
+  if (SrcTy->isFunctionType())
```

Comment:
I don't understand why we are checking the return type of E1 if E1 is a function type.    I think we  do not want to allow assume_bounds_cast or  dynamic_bounds cast to or from function types.  I think this line and the following line can be deleted.